plaso/engine/artifacts_trie.py

```diff
@@ -0,0 +1,304 @@
+      for artifact_name in node.artifacts_names:
+        for artifact_path in self.artifacts_paths.get(artifact_name, []):
+          if self._ComparePathIfSanitized(
+              current_path, path_separator, artifact_path,
+              node.path_separator):
+            matching_artifacts.add(artifact_name)
+          elif glob.has_magic(artifact_path):
+            if self._MatchesGlobPattern(
+                artifact_path, current_path, node.path_separator):
+              matching_artifacts.add(artifact_name)
+
+    if not segments:
+      return
+
+    segment = segments[0]
+    remaining_segments = segments[1:]
+
+    # Handle glob characters in the current segment.
+    for child_segment, child_node in node.children.items():
+      # Compare the sanitized version of the path segment stored in
+      # the tree to the path segment from to the tool output as it
+      # sanitizes path segments before writing data to disk.
+      sanitized_child_segment = path_helper.PathHelper.SanitizePathSegments(
+              [child_segment]).pop()
+      if segment in (child_segment, sanitized_child_segment):
+        # If the child is an exact match, continue traversal.
+        self._SearchTrie(child_node, self._CustomPathJoin(
+            path_separator,
+            current_path, child_segment), remaining_segments, path_separator,
+            matching_artifacts)
+      # If the child is a glob, see if it matches.
+      elif glob.has_magic(child_segment):
+        if self._MatchesGlobPattern(
+                child_segment, segment, child_node.path_separator):
+          self._SearchTrie(child_node, self._CustomPathJoin(
+              path_separator,
+              current_path, segment), remaining_segments, path_separator,
+              matching_artifacts)
+          self._SearchTrie(
+              node,
+              self._CustomPathJoin(
+                  path_separator,
+                  current_path, segment), remaining_segments, path_separator,
+              matching_artifacts)
+
+  def _ComparePathIfSanitized(
+      self, current_path, path_separator, artifact_path,
+      artifact_path_seperator):
+    """Compares a current path with an artifact path, handling sanitization.
+
+    This method checks if the current_path matches the artifact_path,
+    considering that the artifact_path might have been sanitized.
+
+    Args:
+      current_path (str): The current path being checked.
+      path_separator (str): Path separator for the current path.
+      artifact_path (str): The artifact path to compare against.
+      artifact_path_seperator (str): Path separator for the artifact path.
+
+    Returns:
+      bool: True if the current path matches the artifact path or its
+          sanitized version, False otherwise.
+    """
+    artifact_path_segments = self._GetNonEmptyPathSegments(
+        artifact_path, artifact_path_seperator)
+    sanitized_path_segments = path_helper.PathHelper.SanitizePathSegments(
```

Comment:
Why sanitize the path here? doesn't that cause matches that incorrect matches given the sanitation is lossy ?

Reply:
ack, added a comment